src/libtiled/maptovariantconverter.cpp

```diff
@@ -263,15 +270,41 @@ QVariant MapToVariantConverter::toVariant(const Tileset &tileset,
     return tilesetVariant;
 }
 
-QVariant MapToVariantConverter::toVariant(const TemplateGroup &templateGroup, int firstTid) const
+QVariant MapToVariantConverter::toVariant(const TemplateGroup &templateGroup, int firstTid)
 {
     QVariantMap templateGroupVariant;
 
-    templateGroupVariant[QLatin1String("firsttid")] = firstTid;
+    if (firstTid > 0) {
+        templateGroupVariant[QLatin1String("type")] = QLatin1String("templategroup");
```

Comment:
Actually the `type` should be set when no `firstTid` is passed, and doesn't need to be set when it is.